### PR TITLE
fix(orch): persist the delegated item set per round — dev-round-write

### LINKED ISSUE #1230: dev-fix: delegated item set is not persisted, so a respawned agent cannot write a truthful artifact
## Summary

A `dev-fix` round's delegated item set is not persisted anywhere on disk, so a dev agent respawned mid-round cannot write a truthful completion artifact. The orchestrator must re-send the items from its own context; if that context is gone too, the round's receipt is unrecoverable.

## Observed

Session orchestrating HT-1804 (hyprtrade). Fix round 1 delegated 8 review items to a dev agent via the `dev-fix.md` § 2 delegation prompt. The agent applied all 8 to the working tree, backgrounded `tools/validate`, and ended its turn — the documented correct shape (dev SKILL.md § Long-Running Validation).

The tmux server then died. All subagent sessions were lost. The orchestrator respawned the dev agent and pointed it at the same `Round ID` to finish the tail (validate → commit → `dev-return-write`).

The respawned agent reported it could not write the artifact, because the item texts were not recoverable:

- `tmp/workflow-state-<ISSUE>.json` has `fixed_items: []` — that field is written by `dev-fix.md` § 2 step 7, i.e. only AFTER the round is accepted, which is precisely the state a mid-round respawn has not reached.
- No round-scoped item file exists. `tmp/dev-return-<ISSUE>-<ROUND_ID>.json` is the round's OUTPUT and does not exist yet.
- The review JSONs in `tmp/` hold ~48 raw findings across blockers and suggestions. The delegated set was a curated, renumbered subset of 8. Which findings were selected, and what number each was given, exists only in the delegation message.

The agent correctly refused to guess, since `dev-artifact-check --expect-items` validates the item set and a guessed mapping would put fabricated reasoning into the durable record. The orchestrator still had the delegation in context and re-sent it, so the round completed — but that recovery depended on the orchestrator's context surviving, which is not guaranteed by anything in the workflow.

## Why this matters

The workflow already treats mid-round agent loss as a supported path: SKILL.md § Agent Lifecycle prescribes respawn-and-re-delegate, and `dev-fix.md` § 2 step 6's `ok==false` row prescribes a tail-reconciliation nudge that asks the agent to "write your dev-return artifact ... with one `--item` per review item". Both assume the agent still knows the item set. After a respawn it does not, and nothing on disk closes the gap.

The round token (`dev_round_id`) was added to make acceptance deterministic and immune to stale receipts (vstack#776). That guarantee is per-round, but the item set the receipt is checked against is not persisted per-round, so the two halves are not durable to the same failure.

## Suggested fix

Persist the delegated item set when the round is stamped, alongside `dev_round_id` — e.g. `tmp/dev-round-<ISSUE>-<ROUND_ID>.json` written by the orchestrator at `dev-fix.md` § 2 step 5, or a `dev_round_items` key in workflow state scoped by round id. Then:

- a respawned agent reads its own round's items rather than depending on the orchestrator's context;
- the `ok==false` tail-reconciliation nudge becomes self-sufficient;
- `--expect-items` gains an on-disk source of truth rather than a number list typed into the check invocation.

The same gap applies to `review-pr.md` § 4 and § 7, which build the item set and pass it into `dev-fix.md` as `items` context.

## Environment

- harness: Claude Code
- orch skill: installed via vstack, `source: vstack`
- trigger: tmux server death mid-round (host `/tmp` being wiped by an unidentified process)

### PR DESCRIPTION
**Pull request by bmethod**
Fixes #1230 / VST-222. A dev-fix round's delegated item set lived only in the delegation message: a respawned dev agent could not write a truthful completion artifact, the tail-reconciliation nudge assumed orchestrator context that may be gone, and `dev-artifact-check --expect-items` had no on-disk source of truth (observed live: tmux server death mid-round, HT-1804).

- `dev-round-write` persists the curated item set at delegation time under the round-token discipline (filename token + internal `round_id`, same as dev-return). Same-round records are IMMUTABLE — identical rewrite is idempotent, changed content exits 2 (a changed delegation needs a new round id) — so a partial retry can never shrink the authoritative set.
- `dev-artifact-check --expect-items-from-round` derives the expected item set from the record; full schema validation with edge-case tests.
- Real review blocks travel via `--items-file` JSON (orchestrator-side, harness-safe) — backticks/apostrophes/hostile text never ride shell quoting; inline `--item` stays for plain text.
- Workflow docs updated both sides (orch dev-fix/review-pr-comments; dev dev-fix respawn path reads `items[].text` instead of guessing); analysis rounds explicitly carve out. README/DEVELOPMENT document the new script and flag.

Suite: full orch run-all 37/37 bare; cross-model review's three blockers and five suggestions all absorbed.

https://claude.ai/code/session_01L7fubfyViYcy7zUNL6uLPf

### CODE REVIEW

File: skills/orch/scripts/dev-round-write [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+# An identical re-invocation (a retry after an unclear failure) is idempotent;
+# different content under the same token would silently rewrite what was
+# delegated — e.g. a retry with a partial list shrinking the set the
+# completion artifact is checked against. jq output is deterministic, so a
+# byte comparison decides identity.
+if [[ -f "$out" ]]; then
+  if cmp -s "$tmp" "$out"; then
+    rm -f "$tmp"
+    trap - EXIT
+    printf '%s\n' "$out"
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/orch/scripts/dev-round-write]: Fixed in 52256363: placement is now ln(2) create-if-absent — EEXIST is atomic, so exactly one concurrent contender places the record; every other lands in the byte-compare branch (identical retry idempotent, divergent refused, same diagnostics). The race window itself is not unit-instrumentable, but both post-race branches are pinned by the existing immutability tests, which now exercise the atomic path.

File: skills/orch/scripts/dev-round-write [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+# --- Same-round records are immutable ---
+# The record is the authoritative delegated set for exactly one delegation.
+# An identical re-invocation (a retry after an unclear failure) is idempotent;
+# different content under the same token would silently rewrite what was
+# delegated — e.g. a retry with a partial list shrinking the set the
+# completion artifact is checked against. jq output is deterministic, so a
+# byte comparison decides identity.
+if [[ -f "$out" ]]; then
+  if cmp -s "$tmp" "$out"; then
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/orch/scripts/dev-round-write]: Fixed in 52256363: placement is now ln(2) create-if-absent — EEXIST is atomic, so exactly one concurrent contender places the record; every other lands in the byte-compare branch (identical retry idempotent, divergent refused, same diagnostics). The race window itself is not unit-instrumentable, but both post-race branches are pinned by the existing immutability tests, which now exercise the atomic path.

File: skills/orch/scripts/dev-round-write [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    rm -f "$tmp"
+    trap - EXIT
+    printf '%s\n' "$out"
+    exit 0
+  fi
+  die "round record $out already exists with different content — the delegated set is stamped once per round; a changed delegation needs a NEW round id (workflow-state new-round-id)"
+fi
+
+mv "$tmp" "$out"
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/orch/scripts/dev-round-write]: Fixed in 52256363: placement is now ln(2) create-if-absent — EEXIST is atomic, so exactly one concurrent contender places the record; every other lands in the byte-compare branch (identical retry idempotent, divergent refused, same diagnostics). The race window itself is not unit-instrumentable, but both post-race branches are pinned by the existing immutability tests, which now exercise the atomic path.

File: skills/dev/workflows/dev-fix.md [hunk 4]
```diff
@@ -130,6 +130,8 @@ Criteria: Would this save 5+ minutes in a future session? If yes, update. One su
 ```
 
 
+**If you were respawned mid-round and no longer have the `Review items:` list, do not reconstruct it from the raw review JSONs and do not guess** — the delegated set was a curated, renumbered subset, and a guessed mapping puts fabricated reasoning into the durable record. The orchestrator persisted your round's item set at delegation: read `[WORKTREE_PATH]/tmp/dev-round-[ARTIFACT_KEY]-[DEV_ROUND_ID].json` (each `items[]` entry carries the delegated number `n` and the item's full text) and write one `--item` per entry. If that file is missing too, report the gap to the orchestrator instead of writing an artifact you cannot back.
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/dev/workflows/dev-fix.md:133]: Fixed in 52256363: the writer's --issue and the dev-side reader's [ARTIFACT_KEY] are the SAME normalized workflow-state key (Parent ID when bundled) — the equivalence is now stated at the orch call site (dev-fix.md step 5) and in schemas/dev-round.md, so a bundled round cannot strand the record under a key no reader derives. review-pr-comments.md already defines its [ISSUE_ID] as the normalized key at its top.

File: skills/orch/DEVELOPMENT.md [hunk 5]
```diff
@@ -139,6 +139,35 @@ remains, now solely as the stall watchdog deadline.
 - Writes `round_id` and `schema_version: 1`; builds the JSON with `jq` (never string concat) to a same-dir temp file and `mv`s it over the target (atomic — a concurrent checker never sees a partial artifact, and a failed generation leaves any prior receipt intact).
 - Any usage/validation error → stderr + exit 2 (bad `--kind`, missing required arg, malformed `--validate`, path-unsafe `--issue`/`--round-id`, bad `--item` DECISION, empty REASONING, non-integer `N`, a missing or explicitly empty `--summary-file` value, a whitespace-only `--summary`, `--summary` combined with `--summary-file` (presence-based — an empty value still counts as supplied), a single-valued flag supplied twice, a value slot filled by one of the script's own flag tokens (a forgotten value), a `fix`/`--bundled` invocation with no `--item`, or an analysis invocation carrying a rejected flag); on success prints the artifact's absolute path.
 
+### `dev-round-write` — the round's input record (vstack#1230)
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/orch/DEVELOPMENT.md]: Fixed in 52256363: the one citation this branch added to DEVELOPMENT.md is removed, matching the #1246-round precedent for these docs; pre-existing citations in untouched sections stay per that same precedent, and schemas keep provenance per the rule's explicit exemption.
